// auditjs/bin/Application/Logger/Logger.js
"use strict";
/*
 * Copyright 2019-Present Sonatype Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
const log4js_1 = require("log4js");
const os_1 = require("os");
const fs_1 = require("fs");
const path_1 = require("path");
exports.DEBUG = 'debug';
exports.ERROR = 'error';
const logPath = path_1.join(os_1.homedir(), '.ossindex');
const logPathFile = path_1.join(logPath, '.auditjs.combined.log');
log4js_1.addLayout('json', function (config) {
    return function (logEvent) {
        return JSON.stringify(logEvent) + config.separator;
    };
});
log4js_1.configure({
    appenders: {
        auditjs: {
            type: 'file',
            maxLogSize: 2 * 1024 * 1024,
            layout: {
                type: 'json',
                separator: ',',
            },
            filename: logPathFile,
        },
        out: { type: 'stdout' },
        errors: { type: 'logLevelFilter', appender: 'out', level: 'error', maxLevel: 'error' },
    },
    categories: {
        default: {
            appenders: ['errors', 'auditjs'],
            level: 'error',
        },
    },
});
const logger = log4js_1.getLogger('auditjs');
logger.level = exports.DEBUG;
exports.createAppLogger = () => {
    if (!fs_1.existsSync(logPath)) {
        fs_1.mkdirSync(logPath);
    }
};
exports.logMessage = (message, level, ...meta) => {
    if (level == exports.DEBUG) {
        logger.debug(message, ...meta);
    }
    else if (level == exports.ERROR) {
        if (meta && meta[0] && meta[0].stack) {
            logger.error(message, meta[0].stack);
        }
        else {
            logger.error(message, ...meta);
        }
    }
};
exports.shutDownLoggerAndExit = (code) => {
    log4js_1.shutdown((err) => {
        if (err) {
            console.error(err);
            process.exit(1);
        }
        process.exit(code);
    });
};
